src/main/java/com/conveyal/gtfs/model/Stop.java

```diff
@@ -125,4 +202,227 @@ public Iterator<Stop> iterator() {
+            while (stopsReader.readRecord()) {
+                String stopId = stopsReader.get(STOP_ID_FIELD);
+                rows.add(createRow(stopsReader, getStopAreaIds(stopAreasByStopId, stopId)));
+            }
+            return (rows.isEmpty())
+                ? stopsReader
+                : produceCsvPayload(rows, CSV_HEADER_FOR_MERGE);
+        } catch (Exception e) {
+            LOG.error("Error while merging stops", e);
+            // Any issues, return the original stops reader (minus stop areas).
+            return stopsReader;
+        }
+    }
+
+    /**
+     * Get all stop areas matching provided stop id.
+     */
+    private static String getStopAreaIds(Map<String, Set<String>> stopAreasByStopId, String stopId) {
+        return Optional.ofNullable(stopAreasByStopId.get(stopId))
+            .map(areas -> String.join(STOP_AREAS_SEPARATOR, areas))
+            .orElse("");
+    }
+
+    /**
+     * Create a CSV row of original stop fields plus the stop area ids.
+     */
+    private static String createRow(CsvReader stopsReader, String stopAreaIds) throws IOException {
+        String[] fields = new String[CSV_FIELDS.length];
+        for (int i = 0; i < CSV_FIELDS.length; i++) {
+            fields[i] = stopsReader.get(CSV_FIELDS[i]);
+        }
+        return String.join(",", fields) + "," + stopAreaIds;
+    }
+
+    /**
+     * Extract the stop areas from file and group by stop id. This is to allow for easier CRUD by the DT UI.
+     */
+    public static Map<String, Set<String>> groupStopAreaIds(CsvReader csvReader, List<String> errors) {
+        Map<String, Set<String>> stopAreasGroupedByStopId = new HashMap<>();
+
+        try {
+            while (csvReader.readRecord()) {
+                if (!hasExpectedNumberOfColumns(csvReader, errors, 2)) {
+                    continue;
+                }
+                String stopAreaId = csvReader.get(AREA_ID_FIELD);
+                String stopId = csvReader.get(STOP_ID_FIELD);
+                stopAreasGroupedByStopId.computeIfAbsent(stopId, k -> new HashSet<>()).add(stopAreaId);
+            }
+            return stopAreasGroupedByStopId;
+        } catch (IOException e) {
+            return Collections.emptyMap();
+        }
+    }
+
+    /**
+     * Expand the stop area ids and write to zip file.
+     */
+    public static void writeStopAreasToFile(ZipOutputStream zipOutputStream, List<Stop> stops) throws IOException {
+        // Create entry for table.
+        zipOutputStream.putNextEntry(new ZipEntry(STOP_AREAS_FILE_NAME));
+        // Create and use PrintWriter, but don't close. This is done when the zip entry is closed.
+        PrintWriter p = new PrintWriter(zipOutputStream);
+        p.print(packStopAreas(stops));
+        p.flush();
+        zipOutputStream.closeEntry();
+    }
+
+    /**
+     * Write stops to zip file.
+     */
+    public static void writeStopsToFile(ZipOutputStream zipOutputStream, List<Stop> stops) throws IOException {
```

Comment:
Nit: `writeStopsToFile` and `writeStopAreasToFile` look almost the same and could be refactored.